nucypher/blockchain/eth/actors.py

```diff
@@ -656,3 +697,342 @@ def request_evaluation(self, evidence):
+        self.read_onchain_stakes(account=address)
+        for staker in self.__stakers:
+            if staker.checksum_address == address:
+                return staker
+        else:
+            raise self.NoStakes(f"{address} does not have any stakes.")
+
+    def __create_staker(self, password: str = None) -> Staker:
+        """Create a new account and return it as a Staker instance."""
+        if self.funding_power.device:
+            # TODO: More formal check for wallet here?
+            # With devices, the last account is is always an unsed one.
+            new_account = self.blockchain.client.accounts[-1]
+        else:
+            if not password:
+                raise self.ConfigurationError("No password supplied to create new staking account.")
+            new_account = self.blockchain.client.new_account(password=password)
+        self.__accounts.append(new_account)
+        staker = Staker(blockchain=self.blockchain, is_me=True, checksum_address=new_account)
+        return staker
+
+    def create_worker_configuration(self, worker_address: str, password: str, **configuration) -> str:
+        """Generates a worker JSON configuration file for a given staking address."""
+        from nucypher.config.characters import UrsulaConfiguration
+        worker_configuration = UrsulaConfiguration.generate(worker_address=worker_address,
+                                                            password=password,
+                                                            config_root=self.config_root,
+                                                            federated_only=False,
+                                                            provider_uri=self.blockchain.provider_uri,
+                                                            **configuration)
+        return worker_configuration
+
+    #
+    # Actions
+    #
+
+    def set_worker(self, staker_address: str, worker_address: str, password: str = None):
+        self.attach_transacting_power(checksum_address=staker_address, password=password)
+        staker = self.get_active_staker(address=staker_address)
+        result = self.staking_agent.set_worker(staker_address=staker.checksum_address,
+                                               worker_address=worker_address)
+
+        self.to_configuration_file(override=True)
+        return result
+
+    def __prepare_new_staker(self, staker: Staker, amount: NU):
+        """Creates a new ethereum account, then transfers it ETH and NU for staking."""
+        if self.funding_tokens < amount:
+            delta = amount - staker.token_balance
+            message = f"{self.funding_account} Insufficient NU (need {delta} more to transfer a total of {amount})."
+            raise self.ConfigurationError(message)
+        if not self.funding_eth:
+            raise self.ConfigurationError(f"{self.funding_account} has no ETH")
+
+        # Transfer NU
+        self.funding_power.activate()
```

Comment:
didn't we already activate this in the __init__?

Reply:
Yes, however, its possible through normal usage to switch `TransactingPower`s with one belonging to another account, like a `Staker`.  We might be able to remove this with some careful inspection, but this is ensuing that we indeed have the funding power attached.